### PR TITLE
Removed httpd.parking and use haproxy + httpd.dispatcher

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Handle parking directly in haproxy + httpd.dispatcher.
Removed httpd.parking service

# Impacts
Parking flow

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Enhancements
* Parking logic has been moved in haproxy and httpd.dispatcher

### CODE REVIEW

File: go/httpdispatcher/proxy.go [hunk 1]
```diff
@@ -184,37 +202,34 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "httpd.dispatcher: database ip4log prepared statement error: %s", err)
+	}
+
+	p.IP6log, err = p.Db.Prepare("select mac from ip6log where ip = ? AND tenant_id = ?")
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "pfdns: database ip6log prepared statement error: %s", err)
+	}
+
+	p.SecurityEvent, err = p.Db.Prepare("Select count(*) from security_event where security_event.security_event_id='1300003' and mac=? and status='open' AND tenant_id = ?")
```
Comment (julsemaan) [inline go/httpdispatcher/proxy.go]: Can you call this ParkingSecurityEvent to make it more explicit

File: go/httpdispatcher/proxy.go [hunk 2]
```diff
@@ -357,3 +373,126 @@ func (p *passthrough) checkDetectionMechanisms(ctx context.Context, e string) bo
 	}
 	return false
 }
+
+// HasSecurityEvents search for parking security event
+func (p *Proxy) HasSecurityEvents(ctx context.Context, mac string) bool {
+	securityEvent := false
+	var securityEventCount int
+	err := p.SecurityEvent.QueryRow(mac, 1).Scan(&securityEventCount)
```
Comment (julsemaan) [inline go/httpdispatcher/proxy.go]: 1 should be a constant for the default tenant id which will simplify a rewrite for multi-tenancy if we do one

File: go/httpdispatcher/proxy.go [hunk 3]
```diff
@@ -357,3 +373,126 @@ func (p *passthrough) checkDetectionMechanisms(ctx context.Context, e string) bo
...
+	t := time.Now()
+
+	// Pass the context in the request
+	r = r.WithContext(ctx)
+	rp.ServeHTTP(w, r)
+	log.LoggerWContext(ctx).Info(fmt.Sprintln(host, time.Since(t)))
+}
+
+// APICall use to unpark a device
+func (p *Proxy) APICall(ctx context.Context, mac string, ip string) error {
```
Comment (julsemaan) [inline go/httpdispatcher/proxy.go]: Should be called APIUnpark or something like more self-explanatory

File: lib/pf/UnifiedApi.pm [hunk 4]
```diff
@@ -1571,6 +1571,7 @@ sub setup_api_v1_cluster_routes {
     my ($self, $root) = @_;
     my $resource_route = $root->any("/cluster")->to(controller => "Cluster")->name("api.v1.Cluster");;
     $resource_route->any(['GET'] => "/servers")->to(action => "servers")->name("api.v1.Cluster.servers");
+    $resource_route->any(['GET'] => "/config")->to(action => "config")->name("api.v1.Cluster.config");
```
Comment (julsemaan) [inline lib/pf/UnifiedApi.pm]: Belongs in this PR?

File: go/httpdispatcher/proxy.go [hunk 5]
```diff
@@ -357,3 +373,126 @@ func (p *passthrough) checkDetectionMechanisms(ctx context.Context, e string) bo
...
+		TLSHandshakeTimeout:   2 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+
+	t := time.Now()
+
+	// Pass the context in the request
+	r = r.WithContext(ctx)
+	rp.ServeHTTP(w, r)
+	log.LoggerWContext(ctx).Info(fmt.Sprintln(host, time.Since(t)))
```
Comment (julsemaan) [inline go/httpdispatcher/proxy.go]: Can you improve the logging here to say it was proxied and not just logging the host + the time

File: lib/pf/parking.pm [hunk 6]
```diff
@@ -54,12 +54,6 @@ sub park {
             "type"        => "int",
         }]);
     }
-    if(isenabled($Config{parking}{show_parking_portal})){
```
Comment (julsemaan) [inline lib/pf/parking.pm:57]: That check isn't part of the new golang dispatcher. I suggest we just deprecate/remove the parameter

**Comment by extrafu** [comment]
@julsemaan All good for merging?

**Comment by julsemaan** [comment]
Forgot to submit the review when I did it 3 weeks ago, thank god Github remembered it, just submitted the review

File: lib/pf/parking.pm (hunk 6 above)
Comment (fdurand) [reply lib/pf/parking.pm:57]: Added the check in the dispatcher

**Comment by extrafu** [comment]
Ready to be merged?

**Pull request merged**